src/libraries/Microsoft.Extensions.Hosting/src/Internal/ConsoleLifetime.cs

```diff
@@ -79,6 +80,13 @@ private void OnApplicationStarted()
             Logger.LogInformation("Application started. Press Ctrl+C to shut down.");
             Logger.LogInformation("Hosting environment: {EnvName}", Environment.EnvironmentName);
             Logger.LogInformation("Content root path: {ContentRoot}", Environment.ContentRootPath);
+            string contentRootFullPath = Path.GetFullPath(Environment.ContentRootPath);
+
+            if (contentRootFullPath.Equals(System.Environment.CurrentDirectory, StringComparison.Ordinal)
+                || contentRootFullPath.Equals(AppContext.BaseDirectory, StringComparison.Ordinal))
```

Comment:
When are these different?

Reply:
* when DisableDefaults = true, it is System.Environment.CurrentDirectory :

  * code :

https://github.com/dotnet/runtime/blob/d3f7d59e8dc44e5ce22c8ec9c011a1b72d7e9e92/src/libraries/Microsoft.Extensions.Hosting/src/HostingHostBuilderExtensions.cs#L216-L227

  * test :

https://github.com/dotnet/runtime/blob/d3f7d59e8dc44e5ce22c8ec9c011a1b72d7e9e92/src/libraries/Microsoft.Extensions.Hosting/tests/UnitTests/HostApplicationBuilderTests.cs#LL519C9-L524C10

* else it is AppContext.BaseDirectory :

  * code :

https://github.com/dotnet/runtime/blob/d3f7d59e8dc44e5ce22c8ec9c011a1b72d7e9e92/src/libraries/Microsoft.Extensions.Hosting/src/HostBuilder.cs#L230-L234

  * test :

https://github.com/dotnet/runtime/blob/d3f7d59e8dc44e5ce22c8ec9c011a1b72d7e9e92/src/libraries/Microsoft.Extensions.Hosting/tests/UnitTests/HostApplicationBuilderTests.cs#LL251C9-L283C10

Reply:
I'm not sure if there's a case when those two properties are different, but the two seems different by looking at the code